src/pages/home/sidebar/SidebarLinksData.js

```diff
@@ -79,13 +81,14 @@ function SidebarLinksData({isFocused, allReportActions, betas, chatReports, curr
             return reportIDsRef.current;
         }
 
-        // We need to update existing reports only once while loading because they are updated several times during loading and causes this regression: https://github.com/Expensify/App/issues/24596#issuecomment-1681679531
-        // However, if the user is offline, we need to update the reports unconditionally, since the loading of report data might be stuck in this case.
-        if (!isLoading || !reportIDsRef.current || network.isOffline) {
+        // 1. We need to update existing reports only once while loading because they are updated several times during loading and causes this regression: https://github.com/Expensify/App/issues/24596#issuecomment-1681679531
+        // 2. If the user is offline, we need to update the reports unconditionally, since the loading of report data might be stuck in this case.
+        // 3. Changing priority mode to Most Recent will call OpenApp. If there is an existing reports and the priority mode is updated, we want to immediately update the list instead of waiting the OpenApp request to complete
```

Comment:
NAB - in the interests of time I'll still merge this since it isn't a huge deal, but wanted to note.
```suggestion
        // 3. Changing priority mode to Most Recent will call OpenApp. If there is an existing reports and the priority mode is updated, we want to immediately update the list instead of waiting for the OpenApp request to complete
```